system/lib/wasmfs/file.h

```diff
@@ -63,32 +63,16 @@ class File : public std::enable_shared_from_this<File> {
 
   public:
     Handle(std::shared_ptr<File> file) : file(file), lock(file->mutex) {}
-
-    void getStat(struct stat* buf) {
-      buf->st_dev = 1; // ID of device containing file: Hardcode 1 for now, no meaning at the
-      // moment for Emscripten.
-      buf->st_mode = file->mode;
-      // The number of hard links is 1 since they are unsupported.
-      buf->st_nlink = 1;
-      buf->st_uid = file->uid;
-      buf->st_gid = file->gid;
-      buf->st_rdev = 1; // Device ID (if special file) No meaning right now for Emscripten.
-      if (file->is<Directory>()) {
-        buf->st_size = 4096;
-      } else if (file->is<DataFile>()) {
-        buf->st_size = file->usedBytes;
-      } else { // TODO: add size of symlinks
-        buf->st_size = 0;
-      }
-      // The syscall docs state this is hardcoded to # of 512 byte blocks.
-      buf->st_blocks = (buf->st_size + 511) / 512;
-      buf->st_blksize = 1024 * 1024; // Specifies the preferred blocksize for efficient disk I/O.
-      buf->st_atim.tv_sec = file->atime;
-      buf->st_mtim.tv_sec = file->mtime;
-      buf->st_ctim.tv_sec = file->ctime;
-      // The syscall docs state this is hardcoded to # of 512 byte blocks.
-      buf->st_blocks = (buf->st_size + 511) / 512;
-    }
+    auto usedBytes() -> size_t& { return file->usedBytes; }
+    auto usedBytes() const -> const size_t& { return file->usedBytes; }
```

Comment:
Why use the fancy auto+arrow syntax for return types rather than the traditional syntax?

Also, do you plan to use the `const` overloads? If not, perhaps we can remove them.

Reply:
Oh I can change it back, I just thought it was clearer.